src/elmAnalyse.ts

```diff
@@ -0,0 +1,240 @@
+  private analyseSocket: WebSocket;
+  private analyse: ExecutingCmd;
+  private oc: vscode.OutputChannel = vscode.window.createOutputChannel('Elm Analyse');
+  private messageDescriptionsMap : Map<string, string>
+
+  private initSocketClient() {
+    try {
+      const cwd: string = vscode.workspace.rootPath;
+      const config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('elm');
+      const analyseCommand: string = <string>config.get('analyseCommand');
+      const port: string = <string>config.get('analysePort');
+      const wsPath = "ws://localhost:" + port + "/state";
+      if (this.analyseSocket) {
+        this.analyseSocket.close();
+      }
+      this.analyseSocket = new WebSocket(wsPath);
+      this.analyseSocket.on("open", () => {
+        this.analyseSocket.send("PING");
+        this.statusBarInformation.text = 'elm-analyse websocket listening on port ' + port;
+        this.statusBarInformation.show();
+      });
+      this.analyseSocket.on("message", (stateJson) => {
+        try {
+          this.elmAnalyseIssues = [];
+          const state = JSON.parse(stateJson);
+          const messages = state.messages;
+          messages.forEach(message => {
+            if (message.hasOwnProperty("data") && message.data.hasOwnProperty("type")) {
+              let messageType = message.data.type;
+              if (message.data.hasOwnProperty(messageType)) {
+                let messageTypeInfo = message.data[messageType];
+                let messageInfoFileSrc = messageTypeInfo.file;
+                let messageInfoFileRange = [0, 0, 0, 0];
+                if (messageTypeInfo.hasOwnProperty("range")) {
+                  messageInfoFileRange = messageTypeInfo.range.real;
+                }
+                let detail = "";
+                if (this.messageDescriptionsMap.has(messageType)) {
+                  detail = this.messageDescriptionsMap.get(messageType);
+                }
+                let issue: IElmIssue = {
+                  tag: 'analyser',
+                  overview: messageType,
+                  subregion: '',
+                  details: detail,
+                  region: this.correctRange(messageInfoFileRange),
+                  type: 'warning',
+                  file: path.join(cwd, messageInfoFileSrc)
+                };
+                this.elmAnalyseIssues.push(issue);
+              }
+            }
+          });
+          runLinter(vscode.window.activeTextEditor.document, this)
+        }
+        catch (e) {
+          vscode.window.showErrorMessage('Running websocket against Elm-analyse failed. Check if elm-analyse has been configured correctly.');
+          console.log("Socket not open error", e);
+        }
+      });
+      this.analyseSocket.on("error", (e) => {
+        vscode.window.showErrorMessage('Running websocket against Elm-analyse failed. Check if elm-analyse has been configured correctly.');
+        console.log("ERROR", e);
```

Comment:
You seem not to be consistent using single and double quotes. Is this intentionally? A TS thing?